Sources/Sentry/SentryEnvelope.m

```diff
@@ -31,7 +31,8 @@ - (instancetype)initWithId:(nullable SentryId *)eventId
               traceContext:(nullable SentryTraceContext *)traceContext
 {
     SentrySdkInfo *sdkInfo = [[SentrySdkInfo alloc] initWithName:SentryMeta.sdkName
-                                                      andVersion:SentryMeta.versionString];
+                                                      andVersion:SentryMeta.versionString
+                                                     andPackages:[SentryMeta getSdkPackages]];
```

Comment:
`h`: The SDKInfo of the envelope header is the same as the one from the event payload: see

> This can carry the same payload as the [sdk interface](https://develop.sentry.dev/sdk/data-model/event-payloads/sdk/)
in the event payload but can be carried for all events. This means that SDK
information can be carried for minidumps, session data and other submissions. - https://develop.sentry.dev/sdk/data-model/envelopes/#envelope-headers

Now that we touch both SDKInfo for the envelope header and the event payload, I think it would be great to merge the different implementations. To make this easier we could do this in an extra PR before adding the new packages implementation.